BookClub/models/book.py

```diff
@@ -23,4 +24,22 @@ def __str__(self):
 
     def getPublicationYear(self):
         date = self.publicationYear
-        return date.year
+        return date.year    
+
+    def get_s_size(self):
+        file = urllib.request.urlopen(self.imageS)
+        size = file.headers.get("content-length")
+        file.close()
+        return int(size)
+
+    def get_m_size(self):
+        file = urllib.request.urlopen(self.imageM)
```

Comment:
Worst case, what happens when the file can't be reached? how long will our request try to reach the file. Has it had any effect on performance?

Reply:
The file can always be found as it is in the model, and seemingly has no impact on performance.

Reply:
try-catch?

Reply:
In the model, if an image/url is not found, there is a default cover

Reply:
try:
    response = urllib.request.urlopen(url, timeout=10).read().decode('utf-8')
except (HTTPError, URLError) as error:
    logging.error('Data of %s not retrieved because %s\nURL: %s', name, error, url)
except timeout:
    logging.error('socket timed out - URL %s', url)
else:
    logging.info('Access successful.')
    
something like this.

Set a timeout so we don't get bogged down on it just in case.

Reply:
Obviously except should just return the default cover